Install .deb files and try to do conversions on them

almet opened the pull request:
This integrates with our CI to check that the files present in the repository are installable.

See [the related discussion on the dangerzone repo](https://github.com/freedomofpress/dangerzone/issues/829)

A few notes:

1. For ubuntu, the name of the folders are different from the versions we were using, so I introduced a new `version-name` variable.

2. I wasn't able to just `mv` files to the proper directories, but I'm not sure why. The docs say:

> The steps of a job share the same environment on the runner machine, but run in their own individual processes.

After a few tries, I moved to using artefacts instead, which worked.

apyrgio commented:
Looks great Alexis, thanks a lot for tackling it.

> 1. For ubuntu, the name of the folders are different from the versions we were using, so I introduced a new version-name variable.

Cool. In theory you can use the Ubuntu codenames in our script as well (e.g., `focal` instead of `20.04`. But there may be some landmines here and there, so don't fret.

>  2. I wasn't able to just mv files to the proper directories, but I'm not sure why. The docs say:

I think I know what's going on. You're actually moving the symlink, which points to a relative path:

```
lrwxrwxrwx  1 runner docker   38 Jun 14 15:07 dangerzone_0.6.1-1_all.deb -> ../bookworm/dangerzone_0.6.1-1_all.deb
```

So, when you see that the code complains it can't find the file, it's actually the symlink that can't find the file.

I think `cp` would work here.

almet commented:
>     1. For ubuntu, the name of the folders are different from the versions
>        we were using, so I introduced a new version-name variable.
> 
>    Cool. In theory you can use the Ubuntu codenames in our script as well
>    (e.g., focal instead of 20.04. But there may be some landmines here and
>    there, so don't fret.

Heh. Let's keep it like that for now then :-)

>     2. I wasn't able to just mv files to the proper directories, but I'm
>        not sure why. The docs say:
> 
>    I think I know what's going on. You're actually moving the symlink,
>    which points to a relative path:
> lrwxrwxrwx  1 runner docker   38 Jun 14 15:07 dangerzone_0.6.1-1_all.deb -> ../b
> ookworm/dangerzone_0.6.1-1_all.deb
> 
>    So, when you see that the code complains it can't find the file, it's
>    actually the symlink that can't find the file.
> 
>    I think cp would work here.

Thanks for your comment! I unfortunately don't think that it is related to the
symlink, because it would trigger an "invalid file" error. Here I got a
"non-existing file" error.

In my previous commits (each of them triggered a CI run), I copied the file, and
checked it was there (with a `ls` command), and it was indeed the case.

In the following step I got an error, related to the file not being present
(magically disappearing)

My take was to use the artifacts to solve this, which worked, but
feels suboptimal. I'll try to use [this ssh action](https://github.com/
valeriangalliat/action-sshd-cloudflared) to debug (a bit) further.

almet commented:
I finally found what the problem was, connecting with `ssh` proved helpful to detect some bash expending problems in my script.

I've updated the PR accordingly (and squashed all these push-and-try commits). We should be good to merge.

apyrgio commented:
Nice one Alexis, feel free to merge :slightly_smiling_face:
